src/wp-includes/class-wp-theme.php

```diff
@@ -1860,6 +1860,16 @@ public function get_block_patterns() {
 
 		$files = (array) self::scandir( $dirpath, 'php', -1 );
 
+		/**
+		 * Filters list of block pattern files for a theme.
+		 *
+		 * @since 6.8
+		 *
+		 * @param array  $files   Array of theme files found within `patterns` directory.
+		 * @param string $dirpath Path of theme `patterns` directory being scanned.
+		 */
+		$files = (array) apply_filters( 'get_block_patterns_files', $files, $dirpath );
```

Comment:
Casting as an array could hide if someone returns the wrong type which can make it harder to debug. 

```suggestion
		$files = apply_filters( 'get_block_patterns_files', $files, $dirpath );
```